## Problem Statement 1:

Is gender independent of education level? A random sample of 395 people were
surveyed and each person was asked to report the highest education level they
obtained. The data that resulted from the survey is summarized in the following table:

In [1]:
#          High_School Bachelors   Masters  Ph.d.   Total
# Female        60        54         46     41    201
# Male          40        44         53     57    194
# Total         100       98         99     98    395

Question:

Are gender and education level dependent at 5% level of significance? In other words, given the data collected
above,is there a relationship between the gender of an individual and the level of education that they have obtained?

In [2]:
import numpy as np
import pandas as pd
import scipy.stats as stats

In [3]:
male = [40,44,53,57]
female = [60,54,46,41]
High_school=[60,40]
Bachelors = [54,44]
Masters = [46,53]
Phd = [41,57]
marks = male+female
print(marks)
sex=['M','M','M','M','F','F','F','F']
education =['High_school','Bachelors','Masters','Ph.d','High_school','Bachelors','Masters','Ph.d']
DF=pd.DataFrame({"Education":education,"Marks":marks,"Sex":sex})
DF
print(DF)

[40, 44, 53, 57, 60, 54, 46, 41]
     Education  Marks Sex
0  High_school     40   M
1    Bachelors     44   M
2      Masters     53   M
3         Ph.d     57   M
4  High_school     60   F
5    Bachelors     54   F
6      Masters     46   F
7         Ph.d     41   F


In [4]:
male = [40,44,53,57]
female = [60,54,46,41]

High_school = [60,40]
masters = [46,53]
phd = [41,57]
marks = male + female
print(marks)

sex = ['M','M','M','M','F','F','F','F']
education =['High_school','Bachelors','Masters','Ph.d','High_school','Bachelors','Masters','Ph.d']

df = pd.DataFrame({"Education":education,"Marks":marks,"Sex":sex})

[40, 44, 53, 57, 60, 54, 46, 41]


In [5]:
df

Education  Marks Sex
0  High_school     40   M
1    Bachelors     44   M
2      Masters     53   M
3         Ph.d     57   M
4  High_school     60   F
5    Bachelors     54   F
6      Masters     46   F
7         Ph.d     41   F

lets consider null hypothesis  <br>
   H0(null hyppothesis) = the two categorical variables are indepemdant   <br>
    H1(alternate hypothesis) = the two categorical variables are not independant

In [6]:
cross_tab = pd.crosstab([df.Sex,df.Marks],df.Education,margins=True)
# crosstab() function is used to compute a simple cross tabulation of two (or more) factors.


In [7]:
cross_tab

Education  Bachelors  High_school  Masters  Ph.d  All
Sex Marks                                            
F   41             0            0        0     1    1
    46             0            0        1     0    1
    54             1            0        0     0    1
    60             0            1        0     0    1
M   40             0            1        0     0    1
    44             1            0        0     0    1
    53             0            0        1     0    1
    57             0            0        0     1    1
All                2            2        2     2    8

In [8]:
DF1 = pd.crosstab(DF.Sex, DF.Education,DF.Marks, aggfunc="sum",margins=True)
DF1

Education  Bachelors  High_school  Masters  Ph.d  All
Sex                                                  
F                 54           60       46    41  201
M                 44           40       53    57  194
All               98          100       99    98  395

In [9]:
df1 = pd.crosstab(df.Sex,df.Education,df.Marks,aggfunc="sum",margins=True)

In [10]:
df1

Education  Bachelors  High_school  Masters  Ph.d  All
Sex                                                  
F                 54           60       46    41  201
M                 44           40       53    57  194
All               98          100       99    98  395

In [11]:
DF1.columns = ["Bachelors","High_School","Masters","Ph.d.","Genderwise_total"]
DF1.index = ["Female","Male","Combined"]
DF1

Bachelors  High_School  Masters  Ph.d.  Genderwise_total
Female           54           60       46     41               201
Male             44           40       53     57               194
Combined         98          100       99     98               395

In [12]:
df1.columns = ["Bachelors","High_school","Masters","Ph.d","Genderwise_total"]
df1.index = ["Female","Male","Combined"]

In [13]:
df1

Bachelors  High_school  Masters  Ph.d  Genderwise_total
Female           54           60       46    41               201
Male             44           40       53    57               194
Combined         98          100       99    98               395

In [14]:
df2 = df1.iloc[0:2,0:4]
df2

Bachelors  High_school  Masters  Ph.d
Female         54           60       46    41
Male           44           40       53    57

In [15]:
# For a test of independence, we use the same chi-squared formula that we used for the goodness-of-fit test.
# The main difference is we have to calculate the expected counts of each cellin a 2-dimensional table instead of a 1-dimensional table. To get the expected count for a cell, multiply the row total for that cell by the column
# total for that cell and then divide by the total number of observations. We can quickly get the expected counts
# for all cells in the table by taking the row totals and column totals of the table, performing an outer product on them with the np.outer() function and dividing by the number of observations:


In [16]:
df3 = np.outer(df1["Genderwise_total"][0:2],df1.loc["Combined"][0:4])/395.0
df3 = pd.DataFrame(df3)
df3.columns = ["Bachelors","High_School","Masters","Ph.d."]
df3.index = ["female","male"]
df3

Bachelors  High_School    Masters      Ph.d.
female  49.868354    50.886076  50.377215  49.868354
male    48.131646    49.113924  48.622785  48.131646

In [17]:
# Now we will calculate the chi square statistic
# We called the .sum() twice, once to get the column sum and second time to add the sum, returning the sum of entire 2D table


In [18]:
chi_square_stat = (((df3-df2)**2)/df3).sum().sum()
print(chi_square_stat)

0.0


In [19]:
#Find the critical value for 95% confidence and degree of freedom (df) is 3

In [20]:
critical_val = stats.chi2.ppf(q=0.95,df=3)
print(f'Critical value= {critical_val}')

Critical value= 7.814727903251179


In [22]:
# Now to calculate p value

In [23]:
p_val = 1-stats.chi2.cdf(x=chi_square_stat,df=3)
print(f'P value is= {p_val}')

P value is= 1.0


In [24]:
# Use stats.chi2_contingency() function to conduct a test of independence automatically given a frequency table  of 
# observed counts

In [25]:
results = stats.chi2_contingency(observed=df2)
print(results)

(8.006066246262538, 0.045886500891747214, 3, array([[49.86835443, 50.88607595, 50.37721519, 49.86835443],
       [48.13164557, 49.11392405, 48.62278481, 48.13164557]]))


The output shows the chi-square statistic = 8, the p-value as 0.045 and the degrees of freedom as 3
The critical value with 3 degree of freedom is 7.815. Since 8.006 > 7.815, therefore we reject the null hypothesis and conclude that the education level depends on gender at a 5% level of significance.

## Problem Statement 2:
Using the following data, perform a oneway analysis of variance using α=.05. Write up the results in APA format.

[Group1: 51, 45, 33, 45, 67]

[Group2: 23, 43, 23, 43, 45]

[Group3: 56, 76, 74, 87, 56]

In [26]:
#The analysis of variance or ANOVA is a statistical inference test that lets you compare multiple groups at the same time. 
# The one-way ANOVA tests whether the mean of some numeric variable differs across the levels of one categorical variable.
# It essentially answers the question: do any of the group means differ from one another?
#The scipy library has a function for carrying out one-way ANOVA tests called scipy.stats.f_oneway()

In [27]:
group1 = [51, 45, 33, 45, 67]
group2 = [23, 43, 23, 43, 45]
group3 = [56, 76, 74, 87, 56]


# Now to perform Anova Test
statistic,p_value = stats.f_oneway(group1,group2,group3)
print(f'Statistical value= {statistic} P value = {p_value}')
if p_value < 0.05:
    print("Yes, P value is less than 0.05")
else:
    print("No, P value is greater than 0.05")

    

Statistical value= 9.747205503009463 P value = 0.0030597541434430556
Yes, P value is less than 0.05


"The test result suggests the groups have different same sample means in this example, since the p-value is significant at a 99% confidence level. Here the p-value returned is 0.00305 which is < 0.05

## Problem Statement 3:
Calculate F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25. For 10, 20, 30, 40, 50:

In [28]:
stats.f_oneway([10, 20, 30, 40, 50],[5,10,15, 20, 25])

group1 = [10, 20, 30, 40, 50]
group2 = [5,10,15, 20, 25]

mean1 = np.mean(group1)
mean2 = np.mean(group2)

grp1_sub_mean1 = []
grp2_sub_mean2 = []

add1 = 0
add2 = 0

for items in group1:
    add1 += (items - mean1)**2
for items in group2:
    add2 += (items - mean2)**2

var1 = add1/(len(group1)-1)
var2 = add2/(len(group2)-1)

f_test = var1/var2

print(f'Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25 is : {f_test}')


Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25 is : 4.0
